src/publishApprovedTemplate.ts

```diff
@@ -0,0 +1,34 @@
+import { S3 } from 'aws-sdk'
+import * as fs from 'fs';
+
+import {Arguments} from 'yargs';
+import {loadStackArgs} from './cfn/index';
+import configureAWS from './configureAWS';
+
+export async function publishApprovedTemplate(argv: Arguments): Promise<number> {
+  await configureAWS('sandbox', 'us-east-1');
+
+  const stackArgs = await loadStackArgs(argv);
+  const s3        = new S3();
+
+  if (stackArgs.ApprovedTemplateLocation === undefined ||
+      stackArgs.ApprovedTemplateLocation.length < 0 ) {
+    throw new Error("Missing key: `ApprovedTemplateLocation`")
+  }
+
+  var cfnTemplate = await fs.readFileSync(stackArgs.Template)
+  var bucket      = stackArgs.ApprovedTemplateLocation.substr(5).slice(0,-1)
+  var key         = stackArgs.Template.replace("./","")
+  var params      = { Body: cfnTemplate, Bucket: bucket, Key: key, }
+
+  await s3.headBucket({Bucket: bucket}, (err, data) => {
+    if (err) throw new Error(`Invalid Bucket: ${bucket}`)
+  }).promise
+
+  await s3.putObject( params , (err, data) => {
+    if (err) console.log(err, err.stack); // an error occurred
```

Comment:
Haven't thrown anything here yet. I think it would be better to see the whole stack error for now

Reply:
What's the thinking behind using callbacks and promises together? I've never seen this before

Reply:
I am just trying to make it work. I don't know why I needed to use promises. Just saw examples on AWS docs and tried to put it together.

Reply:
OK np!